drivers/hmis_data_quality_tool/spec/models/hmis_data_quality_tool/client_metrics_spec.rb

```diff
@@ -0,0 +1,306 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+require 'rails_helper'
+require_relative './shared_context'
+
+RSpec.describe HmisDataQualityTool::Report, type: :model do
+  include_context 'DQ Tool test setup'
+
+  describe 'Client Metrics' do
+    describe 'Name Issues' do
+      context 'with valid name data' do
+        before do
+          @project = create_project(project_type: 1) # ES
+          @client = create_client_with_warehouse_link(
+            first_name: 'John',
+            last_name: 'Doe',
+          )
+          create_enrollment(client: @client, project: @project)
+          @report = setup_report([@project.id])
+        end
+
+        it 'does not flag valid names' do
+          # Verify report completed successfully
+          expect(@report.state).to eq('Completed')
+          expect(@report.completed_at).not_to be_nil
+
+          # Verify clients are found in the report
+          expect(@report.clients.count).to be > 0
+
+          expect_result(title: 'Name', invalid_count: 0)
+        end
+      end
+
+      context 'with blank name but DQ indicates full name reported' do
+        before do
+          @project = create_project(project_type: 1) # ES
+          @client = create_client_with_warehouse_link(
+            first_name: nil,
+            last_name: 'Doe',
+          )
+          @client.update(NameDataQuality: 1) # Full name reported
```

Comment:
No harm in doing it this way, but I think your updates to `hud_enrollment_builders` should allow this:
```suggestion
          @client = create_client_with_warehouse_link(
            first_name: nil,
            last_name: 'Doe',
            name_data_quality: 1, # Full name reported
          )
```

If not, maybe it should

Reply:
included in [this commit](https://github.com/greenriver/hmis-warehouse/pull/5998/commits/b33d6f972d27cac8e54c392ac5e9cde76fcc1cb9)